c2h/generators.cu

```diff
@@ -509,14 +534,19 @@ INSTANTIATE(double);
 INSTANTIATE(bool);
 INSTANTIATE(char);
 
-#ifdef TEST_HALF_T
+#ifdef _CCCL_HAS_NVFP16
 INSTANTIATE(half_t);
 #endif
 
-#ifdef TEST_BF_T
+#ifdef _CCCL_HAS_NVBF16
 INSTANTIATE(bfloat16_t);
 #endif
 
+#undef INSTANTIATE_RND
+#undef INSTANTIATE_MOD
+#undef INSTANTIATE
+
+#if THRUST_DEVICE_SYSTEM == THRUST_DEVICE_SYSTEM_CUDA
```

Comment:
I believe we should rather just do
```suggestion
#if defined(_CCCL_CUDA_COMPILER)
```

Reply:
Hmm. What if I compile with OMP as device backend but use nvc++ or nvcc as my CXX compiler? The intent here is really to use the same backend as configured for Thrust.

Reply:
Oh yeah that is correct, thanks